### PR TITLE
[Localization] Localize terrain message and translate in Korean

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
Add the option to localize message text of terrain.

## Why am I doing these changes?
All text should be localizable.
+ Weather messages are localized but Terrain aren't.

## What did change?
 - Changed src/data/terrain.ts, src/data/weather.ts for localize
 - Added config and localization key in each language folders.
 - Translated them to Korean.

### Screenshots/Videos
Manual Test of ELECTRIC_TERRAIN
(For longest message test)

| Lang | Start | Clear |
| :-----: | :-----: | :---: |
| En | image | image |
| De | image | image |
| Fr | image | image |
| Pt_BR | image | image |
| Zh_CN | image | image |
| Zh_TW | image | image |
| Ko | image | image |

## How to test the changes?
1. Manually.
 - To manual test with overrides.ts, you can test with moveset override.
```Typescript
export const MOVESET_OVERRIDE: Array<Moves> = [ Moves.MISTY_TERRAIN, Moves.ELECTRIC_TERRAIN, Moves.GRASSY_TERRAIN, Moves.PSYCHIC_TERRAIN ];
```
2. Using test code (src/test/localizaiton/terrain.test.ts)

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/fr/weather.ts [hunk 1]
```diff
@@ -42,3 +42,24 @@ export const weather: SimpleTranslationEntries = {
...
+
+  "grassy": "Grassy",
+  "grassyStartMessage": "Grass grew to cover the battlefield!",
+  "grassyClearMessage": "The grass disappeared from the battlefield.",
+
+  "psychic": "Psychic",
+  "psychicStartMessage": "The battlefield got weird!",
+  "psychicClearMessage": "The weirdness disappeared from the battlefield!",
+
+  "defaultBlockMessage": "{{pokemonNameWithAffix}} is protected by the {{terrainName}} Terrain!"
```
Comment (Adri1) [inline src/locales/fr/weather.ts:64]:
```suggestion
  "misty": "Brumeux",
  "mistyStartMessage": "La brume recouvre le terrain !",
  "mistyClearMessage": "La brume qui recouvrait le terrain se dissipe…",
  "mistyBlockMessage": "La brume enveloppe {{pokemonNameWithAffix}} !",

  "electric": "Électrifié",
  "electricStartMessage": "De l’électricité parcourt le terrain !",
  "electricClearMessage": "L’électricité parcourant le terrain s’est dissipée…",

  "grassy": "Herbu",
  "grassyStartMessage": "Un beau gazon pousse sur le terrain !",
  "grassyClearMessage": "Le gazon disparait…",

  "psychic": "Psychique",
  "psychicStartMessage": "Le sol se met à réagir de façon bizarre…",
  "psychicClearMessage": "Le sol redevient normal !",

  "defaultBlockMessage": "{{pokemonNameWithAffix}} est protégé\npar le Champ {{terrainName}} !"
```

**Review by Adri1 — COMMENTED** [review]
Added French

File: src/locales/de/weather.ts [hunk 2]
```diff
@@ -42,3 +42,24 @@ export const weather: SimpleTranslationEntries = {
...
+
+  "grassy": "Grassy",
+  "grassyStartMessage": "Grass grew to cover the battlefield!",
+  "grassyClearMessage": "The grass disappeared from the battlefield.",
+
+  "psychic": "Psychic",
+  "psychicStartMessage": "The battlefield got weird!",
+  "psychicClearMessage": "The weirdness disappeared from the battlefield!",
+
+  "defaultBlockMessage": "{{pokemonNameWithAffix}} is protected by the {{terrainName}} Terrain!"
```
Comment (CodeTappert) [inline src/locales/de/weather.ts:64]:
```suggestion
  "misty": "Nebelfeld",
  "mistyStartMessage": "Am Boden breitet sich dichter Nebel aus!",
  "mistyClearMessage": "Das Nebelfeld ist wieder verschwunden!",
  "mistyBlockMessage": "{{pokemonNameWithAffix}} wird vom Nebelfeld geschützt!",

  "electric": "Elektrofeld",
  "electricStartMessage": "Elektrische Energie fließt durch den Boden!",
  "electricClearMessage": "Das Elektrofeld ist wieder verschwunden!",

  "grassy": "Grasfeld",
  "grassyStartMessage": "Dichtes Gras schießt aus dem Boden!",
  "grassyClearMessage": "Das Grasfeld ist wieder verschwunden!",

  "psychic": "Psychofeld",
  "psychicStartMessage": "Der Boden fühlt sich seltsam an!",
  "psychicClearMessage": "Das Psychofeld ist wieder verschwunden!",

  "defaultBlockMessage": "{{pokemonNameWithAffix}} wird vom {{terrainName}} geschützt!"
```

File: src/locales/zh_TW/weather.ts [hunk 3]
```diff
@@ -42,3 +42,24 @@ export const weather: SimpleTranslationEntries = {
...
+
+  "grassy": "Grassy",
+  "grassyStartMessage": "Grass grew to cover the battlefield!",
+  "grassyClearMessage": "The grass disappeared from the battlefield.",
+
+  "psychic": "Psychic",
+  "psychicStartMessage": "The battlefield got weird!",
+  "psychicClearMessage": "The weirdness disappeared from the battlefield!",
+
+  "defaultBlockMessage": "{{pokemonNameWithAffix}} is protected by the {{terrainName}} Terrain!"
```
Comment (40chyan) [inline src/locales/zh_TW/weather.ts:64]:
```suggestion
  "misty": "薄霧",
  "mistyStartMessage": "腳下霧氣繚繞！",
  "mistyClearMessage": "腳下的霧氣消失不見了！",
  "mistyBlockMessage": "{{pokemonNameWithAffix}}正受到薄霧場地的保護！",

  "electric": "電氣",
  "electricStartMessage": "腳下電流飛閃！",
  "electricClearMessage": "腳下的電流消失了！",

  "grassy": "青草",
  "grassyStartMessage": "腳下青草如茵！",
  "grassyClearMessage": "腳下的青草消失不見了！",

  "psychic": "精神",
  "psychicStartMessage": "腳下傳來了奇妙的感覺！",
  "psychicClearMessage": "腳下的奇妙感覺消失了！",

  "defaultBlockMessage": "{{pokemonNameWithAffix}}正受到{{terrainName}}的保護！"
```
zh_tw

File: src/locales/zh_CN/weather.ts [hunk 4]
```diff
@@ -42,3 +42,24 @@ export const weather: SimpleTranslationEntries = {
...
+
+  "grassy": "Grassy",
+  "grassyStartMessage": "Grass grew to cover the battlefield!",
+  "grassyClearMessage": "The grass disappeared from the battlefield.",
+
+  "psychic": "Psychic",
+  "psychicStartMessage": "The battlefield got weird!",
+  "psychicClearMessage": "The weirdness disappeared from the battlefield!",
+
+  "defaultBlockMessage": "{{pokemonNameWithAffix}} is protected by the {{terrainName}} Terrain!"
```
Comment (40chyan) [inline src/locales/zh_CN/weather.ts:64]:
```suggestion
  "misty": "薄雾",
  "mistyStartMessage": "脚下雾气缭绕！",
  "mistyClearMessage": "脚下的雾气消失不见了！",
  "mistyBlockMessage": "{{pokemonNameWithAffix}}正受到薄雾场地的保护！",

  "electric": "电气",
  "electricStartMessage": "脚下电光飞闪！",
  "electricClearMessage": "脚下的电光消失不见了！",

  "grassy": "青草",
  "grassyStartMessage": "脚下青草如茵！",
  "grassyClearMessage": "脚下的青草消失不见了！",

  "psychic": "精神",
  "psychicStartMessage": "脚下传来了奇妙的感觉！",
  "psychicClearMessage": "脚下的奇妙感觉消失了！",

  "defaultBlockMessage": "{{pokemonNameWithAffix}}正受到{{terrainName}}的的保护！"
```
zh_cn

Comment (40chyan) [reply src/locales/zh_CN/weather.ts:64]: all based on main game text from abcboy

**Comment by EnochG1** [comment]
Continue at https://github.com/pagefaultgames/pokerogue/pull/2809

**Comment by EnochG1** [comment]
@Adri1 @CodeTappert @40chyan @josericardo-fo 
Our translation may be shown in beta.
If you want to check specific message, I would show test screenshot.

**Comment by CodeTappert** [comment]
> @Adri1 @CodeTappert @40chyan @josericardo-fo 
> Our translation may be shown in beta.
> If you want to check specific message,
> request me then I would show test screenshot.

I mean check if they are too long. Otherwise i think all of us just used the offical text

**Comment by EnochG1** [comment]
@Adri1 @CodeTappert @40chyan @josericardo-fo
Could you please approve PR if screenshots of main article are right?
(You can click and check if images are too small)

**Review by Adri1 — APPROVED** [review]

**Comment by EnochG1** [comment]
@flx-sta,
Please check test code when you have a time.
I tried to use mock i18nnext same as yours.
(I commented wrong PR before. mb)

**Review by CodeTappert — APPROVED** [review]

**Review by josericardo-fo — APPROVED** [review]
LGTM

**Review by 40chyan — APPROVED** [review]
screenshot approved

**Review by mercurius-00 — APPROVED** [review]

**Review by Tempo-anon — APPROVED** [review]
Approved by the wiki/tl team

**Pull request merged**